Microsoft.HealthVault.Fhir/Transformers/ApproximateDateTimeToFhir.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.ItemTypes;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static partial class ItemBaseToFhir
+    {
+        // Register the type on the generic ItemBaseToFhir partial class
+        public static FhirDateTime ToFhir(this ApproximateDateTime approximateDateTime)
+        {
+            return ApproximateDateTimeToFhir.ToFhirInternal(approximateDateTime);
+        }
+    }
+    internal static class ApproximateDateTimeToFhir
+    {
+        internal static FhirDateTime ToFhirInternal(ApproximateDateTime approximateDateTime)
+        {
+            if (!approximateDateTime.ApproximateDate.Month.HasValue && !approximateDateTime.ApproximateDate.Day.HasValue)
+                return new FhirDateTime(approximateDateTime.ApproximateDate.Year);
+            else if (approximateDateTime.ApproximateDate.Month.HasValue && !approximateDateTime.ApproximateDate.Day.HasValue)
+                return new FhirDateTime(approximateDateTime.ApproximateDate.Year, approximateDateTime.ApproximateDate.Month.Value);
+            else if (approximateDateTime.ApproximateDate.Month.HasValue && approximateDateTime.ApproximateDate.Day.HasValue && (approximateDateTime.ApproximateTime == null || !approximateDateTime.ApproximateTime.HasValue))
+                return new FhirDateTime(approximateDateTime.ApproximateDate.Year, approximateDateTime.ApproximateDate.Month.Value, approximateDateTime.ApproximateDate.Day.Value);
+            else
```

Comment:
Can you find a way to simplify these if/else statements it is hard to follow. 

You could probably group them

if(!day){ 
     if(!month){

     } else { 

     }
}

and so on, to make it easy to follow and make sure we do not miss a case when we update it.